docs/guides/latest/workloads/statefulset.md

```diff
@@ -0,0 +1,809 @@
+
+```console
+$ kubectl get pod -n demo
+NAME                READY   STATUS    RESTARTS   AGE
+stash-recovered-0   1/1     Running   0          10m
+stash-recovered-1   1/1     Running   0          11m
+stash-recovered-2   1/1     Running   0          12m
+```
+
+Verify that the sample data has been restored in `/source/data` directory of the `stash-recovered` StatefulSet's pod using the following command,
+
+```console
+$ kubectl exec -n demo stash-recovered-0 -- ls -R /source/data
+/source/data:
+stash-demo-0.txt
+$ kubectl exec -n demo stash-recovered-1 -- ls -R /source/data
+/source/data:
+stash-demo-1.txt
+$ kubectl exec -n demo stash-recovered-2 -- ls -R /source/data
+/source/data:
+stash-demo-2.txt
+```
+
+### Rule Based Restore Process
+
+Generally, Stash runs restore process in all pod's of a StatefulSet. It also provide you what data will be restored into which pod by configuring the `spec.rules`in `BackupSession` crd. You can learn more details from [here](docs/concepts/crds/restoresession.md#specrules)
+
+**Create RestoreSession:**
+
+Now, we are going to create a `RestoreSession` crd targeting the `stash-recovered` StatefulSet to restore the backed up data inside it.
+
+Below is the YAML of the `RestoreSesion` crd that we are going to create,
+
+```yaml
+apiVersion: stash.appscode.com/v1beta1
+kind: RestoreSession
+metadata:
+  name: ss-restore
+  namespace: demo
+spec:
+  driver: Restic
+  repository:
+    name: gcs-repo
+  target:
+    ref:
+      apiVersion: apps/v1
+      kind: StatefulSet
+      name: stash-recovered
+    volumeMounts:
+      - mountPath: /source/data
+        name: source-data
+  rules:
+    - targetHosts: ["host-1","host-2"]
+      sourceHost: "host-1"
+      paths:
+        - /source/data
+    - targetHosts: []
+      sourceHost: ""
+      paths:
+        - /source/data
+```
+
+Here,
+
+- `spec.rules`: `spec.rules` specify how Stash should restore data for each host.
+  - `targetHosts` specify that backed up data of `host-1`(old StatefulSet's pod-1) will be restored in targetHosts `host-1`(new StatefulSet's pod-1) and `host-2`(new StatefulSet's pod-2).
+  - `sourceHost` specify the name of the host whose backed up data will be restored.
+  
+Let's create the `RestoreSession` crd we have shown above,
+
+```console
+$ kubectl apply -f ./docs/examples/guides/latest/workloads/statefulset/adv_restoresession.yaml
+restoresession.stash.appscode.com/ss-restore created
+```
+
+Once, you have created the `RestoreSession` crd, Stash will inject `init-container` to `stash-recovered` StatefulSet. The StatefulSet will restart and the `init-container` will recovered on start-up.
+
+**Verify Init-Container:**
+
+Wait until the `init-container` has been injected to the `stash-recovered` StatefulSet. Let’s describe the StatefulSet to verify that `init-container` has been injected successfully.
+
+```yaml
+$ kubectl describe statefulset -n demo stash-recovered
```

Comment:
Use StatefulSet that has more replicas than original. Then restore data from `pod-0` into the extra replicas.